src/core/ws/WebSocketClientManager.ts

```diff
@@ -28,33 +25,122 @@ export type WebSocketBroadcastChannelMessage =
       }
     }
 
-export const kAddByClientId = Symbol('kAddByClientId')
+type SerializedClient = {
+  clientId: string
+  url: string
+}
 
 /**
  * A manager responsible for accumulating WebSocket client
  * connections across different browser runtimes.
  */
 export class WebSocketClientManager {
+  private inMemoryClients: Set<WebSocketClientConnectionProtocol>
+
+  constructor(
+    private channel: BroadcastChannel,
+    private url: Path,
+  ) {
+    this.inMemoryClients = new Set()
+
+    if (typeof localStorage !== 'undefined') {
+      // When the worker clears the local storage key in "worker.stop()",
+      // also clear the in-memory clients map.
+      localStorage.removeItem = new Proxy(localStorage.removeItem, {
```

Comment:
Ideally, we would know when `worker.stop()` is called but the manager lives in `ws.link()` while the stop event is only ever available in `handleWebSocketEvent()`. 

This would be an insane context drilling. Instead, let's proxy the `localStorage.removeItem()` calls.